### ISSUE TITLE
Randomly started getting an error about an invalid webpack configuration after I added some files to /src/assets/scripts/

### ISSUE THREAD

**Issue by adrianocr** [Shopify/slate#853]
The console error when I run `yarn start` is the following:

```
WebpackOptionsValidationError: Invalid configuration object. Webpack has been initialised using a configuration object that does not match the API schema.
 - configuration.entry should be one of these:
   object { <key>: non-empty string | [non-empty string] } | non-empty string | [non-empty string] | function
   -> The entry point(s) of the compilation.
   Details:
    * configuration.entry should not be empty.
      -> Multiple entry bundles are created. The key is the chunk name. The value can be a string or an array.
    * configuration.entry should be a string.
      -> An entry point without name. The string is resolved to a module which is loaded upon startup.
    * configuration.entry should be an array:
      [non-empty string]
    * configuration.entry should be an instance of function
      -> A Function returning an entry object, an entry string, an entry array or a promise to these things.
    at webpack ([...]/node_modules/webpack/lib/webpack.js:24:9)
    at new DevServer ([...]/node_modules/@shopify/slate-tools/tools/asset-server/index.js:21:21)
    at Promise.all.then ([...]/node_modules/@shopify/slate-tools/cli/commands/start.js:41:19)
    at <anonymous>
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

So I reverted everything I did in the /src/assets/scripts/ directory and the issue persists. 

---

Edit: this is because Slate recently changes. Files are now in /src/scripts/ and /src/styles/. See here: https://github.com/Shopify/slate/blob/master/CHANGELOG.md

**Comment by oscar-dotdev** [comment]
Have encountered the same issue here, can't seem to revert once issue has arisen - even after removing yarn.lock & node_modules folder & reinstalling.

Seems to happen for me when upgrading to newer versions of slate (from beta 6) and then running `yarn install` on an active slate theme.

**Comment by oscarstranger** [comment]
Have encountered the same issue here, can't seem to revert once issue has arisen - even after removing `yarn.lock` & `node_modules` folder & reinstalling.

Seems to happen for me when upgrading to newer versions of slate (from beta 6) and then running `yarn install` on an active slate theme.

**Comment by adrianocr** [comment]
@oscarstranger I just created a new project and from what I can see, everything moved outside of the `assets` dir to individual `styles` and `scripts` directories inside the root of `src`. Investigating now if moving all my files resolves the problem.

**Comment by rainerbird** [comment]
With `v1.0.0-beta.12` upgrade on existing projects you will need to essentially:

- Move your scripts to `src/scripts`
- Move your styles to `src/styles`
- Move your SVGs to `src/snippets` and change extension from `.svg` to `.liquid` 
- No need to keep the `images` and `static` folder inside the `assets` directory anymore, just move all contents of those folders to `src/assets` and then delete those folders. 

More info can be found in the changelog: https://github.com/Shopify/slate/blob/master/CHANGELOG.md

**Comment by oscarstranger** [comment]
@adrianocr @rainerbird Still run into this issue when attempting to migrate from beta 6 to beta 8, so doesn't not seem limited to the file restructure - have tried on multiple versions.

**Comment by rainerbird** [comment]
@oscarstranger if updating to beta 8 as well there is also the breaking update in regards to `slate.config.js`

See under `v1.0.0-beta.8 (2018-09-14)` and here: https://github.com/Shopify/slate/wiki/Slate-Configuration

**Comment by adrianocr** [comment]
@rainerbird I actually got it all to work right before you mentioned that, but thanks for the info nonetheless!

Question though: how are you handling custom fonts now? Before I has an alias in `slate.config.js` `'fonts': path.resolve('./src/assets/fonts/')` and a `fonts` dir inside of `/src/assets/` with all my font files. Then in my `theme.scss` file I referenced each `@font-face` with `~fonts` preceding the font filename. Like this:

```
@font-face {
    font-family: 'Big Caslon';
    src: url('~fonts/subset-BigCaslon-Medium.woff2') format('woff2'), url('~fonts/subset-BigCaslon-Medium.woff') format('woff');
    font-weight: normal;
    font-style: normal;
}
```

The above was working perfectly before. It doesn't now, even if I move the font files directly into the assets dir and change the alias. I'm hoping I don't have to manually insert it into the head with liquid references.
